test/libService/unittest/stoip6/stoip6test.cpp

```diff
@@ -221,3 +221,55 @@ TEST(stoip6_2, test_2_12)
     CHECK(0 == memcmp(hex_addr[i], buf, 16));
 }
 
+/***********************************************************/
+/* Third test group for stoip6_prefix */
+
+const char string_prefix_addr[][40] =
+{
+    "2001:db8::1:0:0:1/64",     // 1
+    "2001::/60",                // 2
+    "::1/48",                   // 3
+    "::/00",                    // 4
```

Comment:
See you're getting -1 here - that shouldn't happen, should be 0. Looks like that prefix_length function is annoying/broken, returning 0 if no prefix.